AprilSylph/XKit-Rewritten
# Panorama: Custom maximum post width; multiple bug fixes

Note: enabling "hide whitespace" when viewing the diff is recommended.

### Description


Adds a preference letting the user customize the maximum post width when Panorama is active. I really, really wanted the preference to actually be the width of the posts themselves rather than e.g. just setting the width of the outside container.

I experimented with and learned a ton about the box model and intrinsic widths and stuff while researching this... and used none of it, because I was totally overcomplicating things thinking the highest-up-the-tree place to set the timeline width was a layer lower than it is and couldn't interact with the `container` flex, uh, container. Oops.

No idea what to set the default width to, honestly.

Resolves #814,  discussion #391, discussion #853

**Edit:** Additionally, this applies styling to some elements that were missing them (native audio blocks and link embeds), correctly applies Panorama expansion to the tag and search pages when they're not in "masonry" views, and hotfixes the embedded blog view exploding; see the later commits.

Resolves half of #937, the tag page portion; blog view panorama is more complicated and I'll address it in a future PR.

Resolves #985.

### Testing steps


 - enable Panorama
 - adjust the new preference; try a blank value, a number (treated as pixels), and a pixel value. (CSS functions like `calc(100vw - 400px)` should work as well).

additionally:

- ensure Panorama works on an audio post and a link post
- ensure Panorama works on e.g. https://www.tumblr.com/tagged/xkit?sort=recent and https://www.tumblr.com/search/xkit/recent (in "list" view)
- ensure Panorama does nothing on e.g. https://www.tumblr.com/tagged/xkit?sort=recent and https://www.tumblr.com/search/xkit/recent in "masonry" view
- hard navigate to https://www.tumblr.com/april and ensure that it doesn't look like this:

<img width="480" src="https://user-images.githubusercontent.com/8336245/219598611-55782d60-bbf7-4671-8a14-8a2a32f3ac1c.png">
